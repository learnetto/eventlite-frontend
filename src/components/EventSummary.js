import React from 'react'
import PropTypes from 'prop-types'
import Col from 'react-bootstrap/Col'
import Card from 'react-bootstrap/Card'
import moment from 'moment'

const EventSummary = props => (
  <Col md={4}>
    <a href={`/events/${props.event.id}`} className="event-card-link">
      <Card className="mb-4 event-card">
        <Card.Img variant="top" src={props.event.image_url} />
        <Card.Body>
          <Card.Text className="event-card-datetime">
            {moment(props.event.start_datetime).format('ddd, MMM DD, YYYY hh:mm A z')}
          </Card.Text>
          <Card.Title>
            {props.event.title}
          </Card.Title>
        </Card.Body>
      </Card>
    </a>
  </Col>
)

Event.propTypes = {
  event: PropTypes.shape({
    title: PropTypes.string.isRequired,
    start_datetime: PropTypes.string.isRequired,
    location: PropTypes.string.isRequired
  })
}

export default EventSummary
