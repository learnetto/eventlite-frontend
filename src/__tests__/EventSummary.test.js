import React from 'react'
import { render } from '@testing-library/react'
import EventSummary from '../components/EventSummary'
import renderer from 'react-test-renderer'

it('renders event title', () => {
  const { getByText } = render(<EventSummary event={{title: 'My Test Event'}} />)
  const title = getByText('My Test Event')
  expect(title).toBeInTheDocument()
})


it('renders event start_datetime', () => {
  const { getByText } = render(<EventSummary event={{start_datetime: new Date('10 September 2020, 10:00 am')}} />)
  const start_datetime = getByText('Thu, Sep 10, 2020 10:00 AM')
  expect(start_datetime).toBeInTheDocument()
})


it('renders event image', () => {
  const { getByRole } = render(<EventSummary event={{image_url: 'event.png'}} />)
  const image = getByRole('img', {src: 'event.png'})
  expect(image).toBeInTheDocument()
})


it("should render an event summary card", () => {
  const tree = renderer
    .create(<EventSummary event={{
      id: 1,
      title: 'My Test Event',
      start_datetime: new Date('10 September 2020, 10:00 am'),
      image_url: 'event.png'
    }} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
