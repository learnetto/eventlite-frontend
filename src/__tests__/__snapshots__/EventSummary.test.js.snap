// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an event summary card 1`] = `
<div
  className="col-md-4"
>
  <a
    className="event-card-link"
    href="/events/1"
  >
    <div
      className="mb-4 event-card card"
    >
      <img
        className="card-img-top"
        src="event.png"
      />
      <div
        className="card-body"
      >
        <p
          className="event-card-datetime card-text"
        >
          Thu, Sep 10, 2020 10:00 AM 
        </p>
        <div
          className="card-title h5"
        >
          My Test Event
        </div>
      </div>
    </div>
  </a>
</div>
`;
